app/src/ble.c

```diff
@@ -312,6 +326,10 @@ int zmk_ble_prof_disconnect(uint8_t index) {
     result = bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
     LOG_DBG("Disconnected from profile %d: %d", index, result);
 
+    raise_profile_changed_event();
+    // `raise_profile_changed_event` actually does `raise_zmk_ble_active_profile_changed`
+    // should `profile_changed` and `active_profile_changed` be two separate events?
+
```

Comment:
```suggestion
```
@petejohanson, should I remove this so that this PR is only the nice!view stuff and minimal core refactor?

That would leave the widget able to display the states correctly whenever it is currently refreshed (ie. on boot, when the _active_ profile changes) but changes on other profiles would not be automatically shown (ie. `BT_DISC`, device disconnects on its own, ...).
 
Having the refresh happen automatically when _any_ profile changes would be left to another PR to raise the appropriate events.

Reply:
Done. This PR should now only contain changes for the nice!view code to track and display the profiles individually, as well as minimal refactoring in `ble.c` and `ble.h` needed to access the profile states individually.